elements/bulbs-liveblog/bulbs-liveblog.test.js

```diff
@@ -0,0 +1,500 @@
+        '<bulbs-liveblog> element MUST specify a `firebase-path` attribute'
+      );
+    });
+
+    it('requires a `firebase-url` attribute', () => {
+      subject.removeAttribute('firebase-url');
+      expect(() => subject.attachedCallback()).to.throw(
+        '<bulbs-liveblog> element MUST specify a `firebase-url` attribute'
+      );
+    });
+
+    it('requires a `firebase-api-key` attribute', () => {
+      subject.removeAttribute('firebase-api-key');
+      expect(() => subject.attachedCallback()).to.throw(
+        '<bulbs-liveblog> element MUST specify a `firebase-api-key` attribute'
+      );
+    });
+
+    it('requires a `liveblog-new-entries-url` attribute', () => {
+      subject.removeAttribute('liveblog-new-entries-url');
+      expect(() => subject.attachedCallback()).to.throw(
+        '<bulbs-liveblog> element MUST specify a `liveblog-new-entries-url` attribute'
+      );
+    });
+
+    it('sets up firebase', () => {
+      subject.attachedCallback();
+      expect(subject.setupFirebase).to.have.been.called.once;
+    });
+
+    it('starts a lookup interval', () => {
+      subject.attachedCallback();
+      expect(subject.setupInterval).to.have.been.called.once;
+    });
+
+    it('registers event handlers', () => {
+      subject.attachedCallback();
+      expect(subject.setupEvents).to.have.been.called.once;
+    });
+
+    it('tracks newEntriesButtons', () => {
+      subject.attachedCallback();
+      expect(subject.newEntriesButtons).to.be.an.instanceof(HTMLCollectionClass);
+    });
+
+    it('creates a staging pad for new entries', () => {
+      subject.attachedCallback();
+      expect(subject.entryStaging.matches('div')).to.be.true;
+    });
+
+    it('hides the staging pad for new entries', () => {
+      subject.attachedCallback();
+      expect(subject.entryStaging.style.display).to.eql('none');
+    });
+
+    it('initializes entriesData', () => {
+      subject.attachedCallback();
+      expect(subject.entriesData).to.eql({});
+    });
+
+    it('tracks the liveblog entries container', () => {
+      subject.attachedCallback();
+      expect(subject.entriesContainer).to.be.an.instanceof(HTMLCollectionClass);
+    });
+
+    it('tracks the liveblog-entries', () => {
+      subject.attachedCallback();
+      expect(subject.entriesElements).to.be.an.instanceof(HTMLCollectionClass);
+    });
+  });
+
+  describe('setupFirebase', () => {
+    xit('creates a firebase ref to the `firebase-path`', () => {
+
+    });
+
+    xit('orders the firebase query by published', () => {
+
+    });
+
+    xit('limits the firebase query to 100 items', () => {
+
+    });
+  });
+
+  describe('setupInterval', () => {
+    it('connects an interval handler every LIVEBLOG_LATENCY seconds', () => {
+      sinon.stub(window, 'setInterval');
+      subject.setupInterval();
+      expect(window.setInterval).to.have.been.calledWith(subject.handleInterval, 5000);
+      window.setInterval.restore();
```

Comment:
If you use a [sandbox](http://sinonjs.org/docs/#sandbox) you can restore in an `afterEach` for the whole suite:

```
var sandbox;

beforeEach(() => {
  sandbox = sinon.sandbox.create();
});

afterEach(() => {
  sandbox.restore();
});

it('should stub setInterval on the window', () => {
  sandbox.stub(window, 'setInterval');
  ...
});

```

Any place you would use `sinon` in the test suite, you now use `sandbox`.

If you get into the habit, it avoids potentially forgetting to restore a global and affecting the behavior of other tests. Also covers the case where you attempt to `restore` after a failed assertion, `restore` will never be called in the current case.